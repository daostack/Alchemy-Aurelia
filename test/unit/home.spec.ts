import { bootstrap } from 'aurelia-bootstrapper';
import { StageComponent } from 'aurelia-testing';
import { PLATFORM } from 'aurelia-pal';

describe('Home Page', () => {
  let component;

  beforeEach(async () => {
    component = StageComponent
      .withResources('home')
      .inView('<home></home>');
    await component.create(bootstrap);
  });

  // only jest supports creating snapshot:
  if (jest) {
    it('should render correctly', () => {
      expect(document.body.outerHTML).toMatchSnapshot();
    });
  }

  it('should render heading', () => {
    const headerElement = document.querySelector('h1') as HTMLElement;
    expect(headerElement).not.toBeFalsy();
    expect(headerElement.innerHTML).toBe('DAOs for an Emergent Future');
  });

  afterEach(() => {
    component.dispose();
  });
});
