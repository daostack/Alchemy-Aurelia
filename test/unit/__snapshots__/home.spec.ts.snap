// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render correctly 1`] = `
"<body><div><home class=\\"au-target\\" au-target-id=\\"1\\">
  <!-- <require from=\\"./resources/customAttributes/blur-image\\"></require> -->

  <div class=\\"home\\">
    <div style=\\"margin-bottom: 2rem\\">
      <h1>Creating DAOs for a Collective Intelligence</h1>
      <h4>Welcome to DAOstack, the world's first open ecosystem of Decentralized Autonomous Organizations!</h4>
      <p>DAOstack is an open source framework for a DAO management platform and ecosystem of organizations, collaborators and
        investors that collaborate on the blockchain.</p>
      <p>Let's get started!</p>
    </div>

    <div class=\\"row\\">
      <div class=\\"col col-sm-12 col-md-6\\">
        <div class=\\"card hoverable hot text-center\\" click-to-route=\\"organizations\\">
          <div class=\\"card-block\\">
            <div class=\\"card-image\\">
              <img class=\\"big-button-image\\" src=\\"../static/Tree-of-Life_Flower-of-Life_Stage.svg\\">
            </div>
            <h4 class=\\"card-title\\">Browse the DAOstack Ecosystem</h4>
          </div>
        </div>
      </div>
      <div class=\\"col col-sm-12 col-md-6\\">
        <div class=\\"card hoverable hot text-center\\" click-to-route=\\"deployDAO\\">
          <div class=\\"card-block\\">
            <div class=\\"card-image\\">
              <img class=\\"big-button-image text-center\\" src=\\"../static/mandala-02-black.svg\\">
            </div>
            <h4 class=\\"card-title\\">Birth a New DAO</h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</home></div></body>"
`;
